src/main/java/org/reaktivity/nukleus/socks/internal/stream/client/AcceptStreamHandler.java

```diff
@@ -0,0 +1,328 @@
+    {
+        acceptHandlerState.accept(msgTypeId, buffer, index, length);
+    }
+
+    @State
+    private void beforeBeginState(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
+    {
+        if (msgTypeId == BeginFW.TYPE_ID)
+        {
+            final BeginFW begin = context.beginRO.wrap(buffer, index, index + length);
+            handleBegin(begin);
+        }
+        else
+        {
+            doReset(acceptThrottle, acceptStreamId);
+        }
+    }
+
+    private void handleBegin(
+        BeginFW begin)
+    {
+        acceptReplyEndpoint = context.router.supplyTarget(acceptSourceName);
+
+        // TODO confirm this is same BeginFW instance as the one used to create the current AcceptStreamHandler
+        final RouteFW connectRoute = resolveTarget(acceptSourceRef, acceptSourceName);
+        final String connectName = connectRoute.target().asString();
+        connectEndpoint = context.router.supplyTarget(connectName);
+        connectTargetRef = connectRoute.targetRef();
+        connectStreamId = context.supplyStreamId.getAsLong();
+        final long connectCorrelationId = context.supplyCorrelationId.getAsLong();
+
+        final Correlation correlation = new Correlation();
+        correlation.acceptCorrelationId(this.acceptCorrelationId);
+        correlation.acceptName(acceptSourceName);
+        correlation.connectStreamId(connectStreamId);
+        correlation.acceptTransitionListener(this);
+        correlation.connectRef(connectTargetRef);
+
+        context.correlations.put(connectCorrelationId, correlation);
+        final BeginFW connectBegin = context.beginRW
+            .wrap(context.writeBuffer, 0, context.writeBuffer.capacity())
+            .streamId(connectStreamId)
+            .source(NUKLEUS_SOCKS_NAME)
+            .sourceRef(connectTargetRef)
+            .correlationId(connectCorrelationId)
+            .extension(e -> e.reset())
+            .build();
+        connectEndpoint.accept(connectBegin.typeId(), connectBegin.buffer(), connectBegin.offset(), connectBegin.sizeof());
+
+        // FIXME confirm this is the way to handle theottling after sending a message on the CONNECT
+        context.router.setThrottle(connectName, connectStreamId, this::handleConnectThrottle);
+        this.acceptHandlerState = this::beforeConnectState;
+    }
+
+    @Override
+    public void transitionToConnectionReady()
+    {
+        // TODO checkState();
+        // this.doWindow(acceptThrottle, acceptStreamId, 1024, 1024); // TODO replace hardcoded values
+        this.acceptHandlerState = this::afterConnectState;
+    }
+
+    @Override
+    public void transitionToAborted()
+    {
+        // TODO
+    }
+
+    @State
+    private void beforeConnectState(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
+    {
+        doReset(acceptThrottle, acceptStreamId);
+    }
+
+    @State
+    private void afterConnectState(
```

Comment:
Isn't the presence of the annotation and the State on the method noise?  probably should have one or the other